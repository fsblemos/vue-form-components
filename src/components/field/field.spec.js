import { shallowMount } from '@vue/test-utils';
import VfInput from './field.vue';

describe('Field Component', () => {
  const shallowComponent = (propsData, { slots } = {}) =>
    shallowMount(VfInput, { propsData, slots });

  it('should render a default field', () => {
    const wrapper = shallowComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render a field with label and input', () => {
    const slots = { default: '<input />' };
    const wrapper = shallowComponent({ label: 'label' }, { slots });
    expect(wrapper.element).toMatchSnapshot();
  });
});
