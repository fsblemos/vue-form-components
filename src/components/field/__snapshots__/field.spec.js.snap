// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Component should render a default field 1`] = `
<div
  class="vf-field"
>
  <!---->
   
</div>
`;

exports[`Field Component should render a field with label and input 1`] = `
<div
  class="vf-field"
>
  <label>
    label
  </label>
   
  <input />
</div>
`;
